### PR TITLE
feat: Update Cohere default LLMs, add examples and update unit tests

### PR DESCRIPTION
**Pull request by vblagoje**
- Update default model to `command-r`
- Add examples

### CODE REVIEW

File: integrations/cohere/examples/cohere_generation.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from typing import List
```
Comment (anakin87) [inline integrations/cohere/examples/cohere_generation.py:1]: I think adding a brief comment (and/or some docs links) explaining this example would be beneficial for users unfamiliar with `BranchJoiner` and `JsonSchemaValidator`.

**Review by anakin87 — APPROVED** [review]
Looks good.

Please just add an explanatory comment in the generation example.

File: integrations/cohere/examples/cohere_generation.py (hunk 1 above)
Comment (vblagoje) [reply integrations/cohere/examples/cohere_generation.py:1]: Deal, will do

Comment (vblagoje) [reply integrations/cohere/examples/cohere_generation.py:16]: Have a brief look at [4b1bee8](https://github.com/deepset-ai/haystack-core-integrations/pull/838/commits/4b1bee8df99d01d5fc1620d2c8a18391a33efda5) @anakin87

Comment (anakin87) [reply integrations/cohere/examples/cohere_generation.py:16]: Nice!

**Pull request merged**